### PR TITLE
Make the dot-colon method reference frozen

### PR DESCRIPTION
**Pull request by mensfeld**
A follow-up to my discussion with @ko1. This PR freezes the method reference when fetched using the new `dot-symbol` syntax sugar.

It's worth keeping the bound methods frozen as one should not modify them. Freezing keeps also a window of opportunity for introducing method reference caching in case it would be needed due to the fact, that the method object is immutable.

### CODE REVIEW

**Comment by mensfeld** [comment]
The `appveyor` failures are not related to this PR but related to @davydovanton logger changes that were merged before this PR was introduced.

File: test/ruby/test_method.rb [hunk 1]
```diff
@@ -1136,4 +1136,17 @@ def o.method(m); nil; end
     assert_equal(m, o.:foo)
     assert_nil(o.method(:foo))
   end
+
+  def test_method_reference_freeze_state
+    m = 1.:succ
+    assert(m.frozen?)
```
Comment (nobu) [inline test/ruby/test_method.rb]: `assert_predicate` will be better for the failure message.

Comment (mensfeld) [reply test/ruby/test_method.rb]: @nobu done.

Comment (nobu) [reply test/ruby/test_method.rb]: and `assert_not_predicate` instead of `assert_equal(..., false)`.

Comment (mensfeld) [reply test/ruby/test_method.rb]: @nobu fixed again <3

**Comment by mensfeld** [comment]
@nobu no rush but if this would be merged, I could open a proposal for the `.:` last used method cache.

**Comment by nobu** [comment]
Please file at [New Issue](https://bugs.ruby-lang.org/projects/ruby-trunk/issues/new), and add it to [DevelopersMeeting20190829Japan](https://bugs.ruby-lang.org/issues/15996).
Then we'll discuss it at the next meeting.

**Comment by mensfeld** [comment]
@nobu  https://bugs.ruby-lang.org/issues/16103 - I don't feel competent enough t add it to the developers meeting and me dunno how so I would appreciate if you could drop it there. Thank you in advance!

**Comment by mensfeld** [comment]
@nobu I did a rebase, the fails seem not related. I believe, this tests still should be merged.

**Comment by nobu** [comment]
Sorry, I've forgotten this PR.
And a failure in test-bundler should be cleared by rebasing.

**Pull request merged**